Source/WebKit/UIProcess/ios/CompactContextMenuPresenter.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if USE(UICONTEXTMENU)
+
+#import <UIKit/UIKit.h>
+#import <wtf/RetainPtr.h>
+
+@class WKCompactContextMenuPresenterButton;
+
+namespace WebKit {
+
+class CompactContextMenuPresenter {
+    WTF_MAKE_NONCOPYABLE(CompactContextMenuPresenter); WTF_MAKE_FAST_ALLOCATED;
+public:
+    CompactContextMenuPresenter(UIView *rootView, id<UIContextMenuInteractionDelegate>);
+    ~CompactContextMenuPresenter();
+
+    void present(CGRect rectInRootView);
+    void present(CGPoint locationInRootView);
+    void dismiss();
+
+    UIContextMenuInteraction *interaction() const;
+
+private:
+    __weak UIView *m_rootView { nil };
```

Comment:
is there a reason to prefer this over using a `WeakObjCPtr`?

Reply:
It's cleaner than `WeakObjCPtr` (the latter was only necessary in WebKit2 before it was [enabled it a few years ago](https://commits.webkit.org/200302@main))